pkg/process/collector.go

```diff
@@ -0,0 +1,121 @@
+package process
+
+import (
+	"bufio"
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+)
+
+type Options struct {
```

Comment:
I suggest removing Options, to simplify the code, and hardcode "/proc/self/maps" as file to read. Mounting procfs to other place breaks so many other things that we can safely assume this.

Reply:
This was actually done for unit tests. We don't let Cortex users to configure it, but having options looks an easy way to mock `/proc` for unit tests. Same for the pid.

Reply:
Another (simpler?) option would be to pass those two files for testing. This is just a non-blocking nit.